deploy/helm-chart/templates/gateway.yaml

```diff
@@ -0,0 +1,45 @@
+{{- if .Values.gateway.create }}
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: {{ default (include "nginx-gateway.fullname" .) .Values.gateway.name }}
+  namespace: {{ default .Release.Namespace .Values.gateway.namespace }}
+  labels:
+    {{- include "nginx-gateway.labels" . | nindent 4 }}
+  {{- with .Values.gateway.annotations }}
+  annotations:
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  gatewayClassName: {{ .Values.nginxGateway.gatewayClassName }}
+  listeners:
```

Comment:
I wonder why can't we just include listeners as we include ports here https://github.com/nginxinc/nginx-gateway-fabric/blob/f33db51fc9e05ccf98fc8cdae100772a5cc6775e/deploy/helm-chart/values.yaml#L128-L136 ? What's the value of defining the fields of listeners?